src/vxworks.rs

```diff
@@ -11,17 +11,17 @@ use crate::error::{Error, ERRNO_NOT_POSITIVE};
 use core::num::NonZeroU32;
 
 extern "C" {
-    fn randBytes (buf: *mut u8, length: i32) -> i32;
+    fn randBytes(buf: *mut u8, length: i32) -> i32;
     // errnoLib.h
     fn errnoGet() -> i32;
 }
 
 pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
     // Prevent overflow of i32
     for chunk in dest.chunks_mut(i32::max_value() as usize) {
-        let ret = randBytes(chunk.as_mut_ptr(), chunk.len() as i32);
+        let ret = unsafe { randBytes(chunk.as_mut_ptr(), chunk.len() as i32) };
         if ret == -1 {
```

Comment:
Yes: "ERROR" (-1) is returned on error. Note that this can happen even in "normal use" if there isn't enough entropy accumulated. Is that ok?

Reply:
Not, quite. On Linux and other *NIXes we block in such cases until the entropy pool gets properly initialized. Do you have a dedicated error code to signalize "not enough entropy" error?

Reply:
I need to check with our "random" expert to make sure that this is the right function to call for our purposes here (I think maybe randABytes would be better), and how to detect "not enough entropy".

Reply:
To clarify: we want to block if entropy pool has not initialized (i.e. right after boot), but after that we want to draw entropy even if entropy estimate is low (i.e. we don't want to block like reading from `/dev/random` does on Linux).

Reply:
OK in that case I *think* randUBytes would make more sense:

The documentation for that function says:

> This routine is same as randBytes() but it returns OK even if enough entropy has not been created.

But let's wait till I've heard back from our expert, I don't want to mislead you.

Reply:
I discussed this with our randomness expert and the conclusion was:

1. Use `randABytes`

> That API will return error if not enough entropy has been collected for the random state to be considered secure. Once that point has been reached, the API will always succeed, regardless of how much random data has been fetched by calling randBytes/randABytes/randUBytes. 

2. Use `randSecure()` to determine whether to give a fatal error or retry after a sleep

```
< 0 : fatal error
== 0: retry, still gathering entropy
> 0: ok
```

Reply:
Great! I will prepare an appropriate update after the `libc` PR lands on crates.io.

Reply:
BTW can we assume that all VxWorks targets have atomics? It would be nice to reuse [`unsync_init`](https://github.com/rust-random/getrandom/blob/master/src/util.rs#L44) here.

Reply:
Looking at [core::sync::atomic](https://github.com/rust-lang/rust/blob/master/src/libcore/sync/atomic.rs) it appears that Rust atomics are required to be lock-free (whereas in C11 we're allowed to synthesize atomic operations that don't map to hardware operations using a mutex).

unsync_init appears to specifically require atomic load/store for usize.

The targets we plan to support are Intel (32- and 64-bit), ARM (32-bit and 64-bit), and (eventually) PowerPC (32- and 64-bit). From what I can tell these all support atomic operations on usize.

So in short, yes, we **can** use unsync_init here. And thanks for the question, I learned something in answering it!

Reply:
Will `randSecure()` report error code via `errnoGet()` or directly in return code?

Reply:
Also, is it guaranteed that if `randABytes` has returned non-negative value, then the buffer got filled completely or does it return how many bytes have been generated?

Reply:
> Will `randSecure()` report error code via `errnoGet()` or directly in return code?

errno is not set. All status is reported by the return value, and you can't use that return value for anything other than to distinguish the three cases I mentioned:

< 0 : fatal error
== 0: retry, still gathering entropy
> 0: ok

Reply:
> Also, is it guaranteed that if `randABytes` has returned non-negative value, then the buffer got filled completely

Yes (based on the documentation and a cursory scan of the source code).

>  or does it return how many bytes have been generated?

No, the only cases are:

< 0 : failed
== 0 : success (all bytes written)